File: Shared/src/Models/Domain/StreamMetadata.cs
using System;
using System.Collections.Generic;

namespace Shared.Models.Domain;

public class StreamMetadata
{
    public Guid Id { get; set; }
    
    public Guid StreamId { get; set; }
    
    public string? ThumbnailUrl { get; set; }
    
    public int Views { get; set; }

    public bool IsLive { get; set; } = false;
    
    // Navigation property
    public virtual LiveStream Stream { get; set; } = null!;
}

File: Client/src/app/models/stream/stream-metadata.ts
export class LiveStream {
    id: string;
    streamId: string;
    thumbnailUrl: string;
    views: number;
    isLive: boolean;

    constructor(id: string, streamId: string, thumbnailUrl: string, views: number, isLive: boolean) {
        this.id = id;
        this.streamId = streamId;
        this.thumbnailUrl = thumbnailUrl;
        this.views = views;
        this.isLive = isLive;
    }
}